assumerole.tf

```diff
@@ -0,0 +1,36 @@
+locals {
+  external_id = coalesce(var.assumerole_role_external_id, random_password.external_id.result)
+}
+
+resource "local_file" "assumerole_addrole" {
+  count = var.create_assumerole_template ? 1 : 0
+
+  filename = "assumerole/addrole.sh"
+
+  content = templatefile("${path.module}/template/addrole.sh.tftpl", {
+    stack_name = var.assumerole_stack_name
+  })
+
+}
+
+resource "local_sensitive_file" "assumerole_tfassumerole" {
+  count = var.create_assumerole_template ? 1 : 0
+
+  filename = "assumerole/tfassumerole.cfn.yml"
+
+  content = templatefile("${path.module}/template/tfassumerole.cfn.yml.tftpl", {
```

Comment:
why use CFN for this? why not more TF?

Reply:
there's no chicken/egg problem here. This is for cross account scenarios only and is not intended to hold the state of the account holding this bucket. In single account scenarios and for the "parent" or "delegated" account that will hold tfstate, the CF Stack is the preferred option.